### PR TITLE
Deploy RHODS, LDAP, S3/Minio and run ODS-CI

### PR DESCRIPTION
**Pull request by kpouget**
0. build ODS-CI image (reusing Mustafa's automation work)
1. deploy and configure an LDAP server (François' findings + automation to configure the number of users and have secure admin&user passwords)
2. deploy and configure Minio S3 backend (secured, reusing the LDAP password for authentication)
3. deploy RHODS operator (mostly reusing `ci-artifacts` generic operator deployment)
4. launch N=5 Pods with ODS-CI image (not running anything yet though) and collect their execution logs via the cluster's S3 + `oc cp` into the [test artifacts](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/354/pull-ci-openshift-psap-ci-artifacts-master-test-commit/1519384107858006016/artifacts/test-commit/test-commit/artifacts/ods_ci-1/004__rhods__test_jupyterlab/ods-ci/)

(see the CI results [there](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/354/pull-ci-openshift-psap-ci-artifacts-master-test-commit/1519384107858006016/artifacts/test-commit/test-commit/artifacts/ods_ci-1/))

test-path: ods ci-1

### CODE REVIEW

**Comment by kpouget** [comment]
cc @fcami

File: roles/rhods_deploy_ldap/files/ldapgen.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import types
```
Comment (fcami) [inline roles/rhods_deploy_ldap/files/ldapgen.py:1]: This looks strangely familiar. Thanks for making it better.

**Comment by kpouget** [comment]
@fcami PR starts to look good from my POV,
[simple test](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/354/pull-ci-openshift-psap-ci-artifacts-master-test-commit/1518980377421549568/artifacts/test-commit/test-commit/artifacts/recipes_test_ods/) is passing (although `ods-ci` has 9/9 failing tests, but that's expected at the moment, it's unhappy because there's no `kubeconfig`, we need to see how to solve this issue)

File: roles/cluster_deploy_minio_s3_server/files/deploy_minio.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+              | _mc admin user add "$MINIO_SERVER_NAME"
+
+           # give privileges to the user
+           _mc admin policy set "$MINIO_SERVER_NAME" consoleAdmin "user=$MINIO_S3_ACCESS_KEY"
+
+           # create default bucket
+           _mc mb minio/mybucket
+
+           echo "All done, Minio is configured!"
+           sleep +inf
```
Comment (jmencak) [inline roles/cluster_deploy_minio_s3_server/files/deploy_minio.yaml:87]: Just out of curiosity, why the `+`?  This is the first time I'm seeing this.  `sleep inf` works just fine for me.

Comment (kpouget) [reply roles/cluster_deploy_minio_s3_server/files/deploy_minio.yaml:87]: to make it clear that we don't wait for `-inf` ;-)

no reason in particular, just like the way it reads

(and the `man` page don't talk about any of these forms :#)

**Comment by kpouget** [comment]
`ci/prow/test-commit` succeeded, see the bottom of [this file](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/354/pull-ci-openshift-psap-ci-artifacts-master-test-commit/1519625407458971648/artifacts/test-commit/test-commit/build-log.txt)

/override ci/prow/test-commit 

`Run ansible-lint on the playbooks and roles / build (pull_request)` failure is irrelevant, I don't know how to mute it

good for me for merging if there's no concern

/approve

**Comment by kpouget** [comment]
/override ci/prow/test-pr 

it's a flake, and `ci/prow/test-commit` already tested this PR, no need to run it again (until something changes)

**Comment by kpouget** [comment]
rebased against master

I'll merge later this morning if the test succeeds

/test test-pr

**Comment by kpouget** [comment]
PR [test](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/354/pull-ci-openshift-psap-ci-artifacts-master-test-pr/1523589533856174080/artifacts/test-pr/test-pr/artifacts/ods_ci-1/004__rhods__test_jupyterlab/) succeeded (although the JupiterHub test itself failing), I'll merge this PR and I'll open another to continue the work

**Pull request merged**